contracts/BucketAuction.sol

```diff
@@ -95,6 +99,37 @@ contract BucketAuction is IBucketAuction, ERC721M {
         return _userData[user];
     }
 
+    function getUserDataPage(uint256 limit, uint256 offset)
+        external
+        view
+        returns (
+            User[] memory,
+            address[] memory,
+            uint256 total
+        )
+    {
+        uint256 numUsers = _users.length();
+        uint256 pageSize = limit;
+        if (pageSize > numUsers - offset) {
```

Comment:
could be `numUsers - offset` underflow?

Reply:
If offset > numUsers, we should simply return empty.

Reply:
i think its fine to let it underflow since this is just a view function, the caller can be responsible for providing valid inputs